### PR TITLE
Fix purecap compilation

### PR DESCRIPTION
**Pull request by arichardson**

### CODE REVIEW

File: machine/vm.h [hunk 1]
```diff
@@ -11,25 +11,27 @@
 # define SATP_MODE_CHOICE INSERT_FIELD(0, SATP64_MODE, SATP_MODE_SV39)
 # define VA_BITS 39
 # define GIGAPAGE_SIZE (MEGAPAGE_SIZE << RISCV_PGLEVEL_BITS)
+typedef uint64_t addr_t;
 #else
 # define SATP_MODE_CHOICE INSERT_FIELD(0, SATP32_MODE, SATP_MODE_SV32)
 # define VA_BITS 32
+typedef uint32_t addr_t;
```
Comment (jrtc27) [inline machine/vm.h:18]: This should probably be `vaddr_t`, since SV32 maps 32-bit virtual addresses to 34-bit physical addresses.

**Comment by arichardson** [comment]
Since pk isn't actually needed I guess we can ignore most of these changes.

**Comment by jrtc27** [comment]
Yeah, I think the `.size`s are the only useful thing (I was just being lazy and ignoring the warnings since we don't really care too much about bbl, as long as it works).

**Comment by bsdjhb** [comment]
We should probably just revert the pk changes I committed previously and leave pk alone entirely.

**Pull request closed without merging**